core/src/main/java/io/undertow/io/UndertowOutputStream.java

```diff
@@ -80,11 +81,9 @@ public void resetBuffer() {
         if(anyAreSet(state, FLAG_WRITE_STARTED)) {
             throw UndertowMessages.MESSAGES.cannotResetBuffer();
         }
-        if(pooledBuffer != null) {
-            pooledBuffer.close();
-            pooledBuffer = null;
```

Comment:
This only nulls the `PooledByteBuffer`, however access using the `buffer()` utility method checks that `buffer` is not null, which this method does not clear, resulting in UndertowOutputStream using a buffer that has potentially been acquired by something else!

I'd really appreciate review when you have a moment @stuartwdouglas, thanks!